src/industry_gui.cpp

```diff
@@ -1262,14 +1290,11 @@ static bool CDECL CargoFilter(const Industry * const *industry, const std::pair<
 			break;
 
 		case CF_NONE:
-			produced_cargo_matches = std::all_of(std::begin((*industry)->produced_cargo), std::end((*industry)->produced_cargo), [](CargoID cargo) {
-				return !IsValidCargoID(cargo);
-			});
+			produced_cargo_matches = std::all_of(std::begin((*industry)->produced), std::end((*industry)->produced), [](const auto &p) { return !IsValidCargoID(p.cargo); });
 			break;
 
 		default:
-			const auto &pc = (*industry)->produced_cargo;
-			produced_cargo_matches = std::find(std::begin(pc), std::end(pc), produced_cargo) != std::end(pc);
+			produced_cargo_matches = std::all_of(std::begin((*industry)->produced), std::end((*industry)->produced), [&produced_cargo](const auto &p) { return p.cargo == produced_cargo; });
```

Comment:
```suggestion
			produced_cargo_matches = std::any_of(std::begin((*industry)->produced), std::end((*industry)->produced), [&produced_cargo](const auto &p) { return p.cargo == produced_cargo; });
```
I think this should be `any_of`, right?